File: evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
from collections import deque
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operators = ["+", "-", "*", "/"]
        stack = deque()
        for token in tokens:
            if token not in operators:
                stack.append(int(token))
            else:
                rightOperand = int(stack.pop())
                leftOperand = int(stack.pop())

                if (token == "+"):
                    result = leftOperand + rightOperand
                elif (token == "-"):
                    result = leftOperand - rightOperand
                elif (token == "*"):
                    result = leftOperand * rightOperand
                else:
                    result = leftOperand / rightOperand 
                    result = floor(result) if result >= 0 else ceil(result)
                
                stack.append(result)
        
        result = stack.pop()
        return result 
            

File: evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        List<String> operators = Arrays.asList("+", "-", "*", "/");
        Stack<Integer> stack = new Stack<>();

        // Iterate through tokens
        for (String token : tokens) {
            if (operators.contains(token)) {
                int rightOperand = stack.pop();
                int leftOperand = stack.pop();

                int result;
                if (token.equals("+")) {
                    result = leftOperand + rightOperand;
                }
                else if (token.equals("-")) {
                    result = leftOperand - rightOperand;
                }
                else if (token.equals("*")) {
                    result = leftOperand * rightOperand;
                }
                else {
                    result = leftOperand / rightOperand;
                }
                stack.push(result);
            }
            else {
                stack.push(Integer.parseInt(token));
            }
        }

        return stack.pop();
    }
}